From 9c8bfb33ab7a52d9cdc178a1e1e36bb0d87a5d98 Mon Sep 17 00:00:00 2001
From: Vicentiu Galanopulo <vicentiu@balena.io>
Date: Mon, 28 Sep 2020 13:47:39 +0200
Subject: [PATCH] Change compatible dts property for watchdog1

The current driver for the imx8 (fsl,imx21-wdt) reboots
the board after a period of time, after issueing
 $echo freeze > /sys/power/state
On BalenaOS this happens due to an active systemd watchdog
which should sleep during deep sleep.
With the fsl,imx8-wdt driver, the watchdog is properly configured
and the reboot does not happen

Upstream-status: Inappropriate [configuration]
Signed-off-by: Vicentiu Galanopulo <vicentiu@balena.io>
---
 arch/arm64/boot/dts/compulab/compulab-imx8mq.dtsi | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm64/boot/dts/compulab/compulab-imx8mq.dtsi b/arch/arm64/boot/dts/compulab/compulab-imx8mq.dtsi
index 711ebdf6f72f..e1f30f4b6201 100644
--- a/arch/arm64/boot/dts/compulab/compulab-imx8mq.dtsi
+++ b/arch/arm64/boot/dts/compulab/compulab-imx8mq.dtsi
@@ -694,6 +694,7 @@
 };
 
 &wdog1 {
+	compatible = "fsl,imx8-wdt";
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_wdog>;
 	fsl,ext-reset-output;
-- 
2.17.1

